subsys/net/lib/app/net_app.c

```diff
@@ -974,6 +974,48 @@ int net_app_close(struct net_app_ctx *ctx)
 	return 0;
 }
 
+int net_app_close2(struct net_app_ctx *ctx, struct net_context *net_ctx)
+{
+	if (!ctx || !net_ctx) {
+		return -EINVAL;
+	}
+
+	if (!ctx->is_init) {
+		return -ENOENT;
+	}
+
+	if (ctx->cb.close) {
```

Comment:
In the "net: app: Add support for multiple listening connections" patch we reversed the order of the close callback and the context clean up.  In that case there was an incoming packet==NULL notification which means we've lost the connection in the lower layers.  Here we are driven by some logic in the application to close the connection, but the different order is still concerning.

Reply:
That is actually excellent point, I will change the cleanup in _net_app_received()